interop/jvm/src/main/scala/scalaz/zio/interop/catsjvm.scala

```diff
@@ -66,31 +159,31 @@ private class CatsEffect extends CatsMonadError[Throwable] with Effect[Task] wit
       try {
         thunk
       } catch {
-        case NonFatal(e) => IO.fail(e)
+        case e: Throwable => IO.fail(e)
       }
     )
 
+  override def delay[A](thunk: => A): Task[A] =
+    IO.syncThrowable(thunk)
+
   override def bracket[A, B](acquire: Task[A])(use: A => Task[B])(
     release: A => Task[Unit]
-  ): Task[B] = IO.bracket(acquire)(release(_).catchAll(IO.terminate(_)))(use)
+  ): Task[B] =
+    IO.bracket(acquire)(release(_).catchAll(IO.terminate))(use)
```

Comment:
`io.catchAll(IO.terminate)` is so common we could do a method to implement that pattern: `io.orDie: IO[Nothing, A]`. This way we could make it zero cost. I'll open a ticket if you like the idea.

Reply:
I like the idea! I've had multiple questions wrt 'how do I make IO[Nothing, ?] from IO[Throwable, ?]'. I think the name might benefit from being long and explicit, e.g. `io.dieOnThrowable`